src/components/PeopleTable.tsx

```diff
@@ -1,645 +1,197 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+import cn from 'classnames';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { PersonLink } from './PersonLink';
+import { Person } from '../types';
+import { getSearchWith } from '../utils/searchHelper';
+
+type Props = {
+  data: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({ data }) => {
+  const { personSlug } = useParams();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const selectedPersonSlug = personSlug ? personSlug : null;
+
+  const sortBy = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  function handleSortField(sortField: string) {
+    if (sortBy !== sortField) {
+      const newSearchParams = getSearchWith(searchParams, {
+        sort: sortField || null,
+      });
+
+      setSearchParams(newSearchParams);
+    } else if (order !== 'desc') {
+      const newSearchParams = getSearchWith(searchParams, {
+        order: 'desc',
+      });
+
+      setSearchParams(newSearchParams);
+    } else {
+      const newSearchParams = getSearchWith(searchParams, {
+        sort: null,
+        order: null,
+      });
+
+      setSearchParams(newSearchParams);
+    }
+  }
+
   return (
-    <table
-      data-cy="peopleTable"
-      className="table is-striped is-hoverable is-narrow is-fullwidth"
-    >
-      <thead>
-        <tr>
-          <th>
-            <span className="is-flex is-flex-wrap-nowrap">
-              Name
-              <a href="#/people?sort=name">
-                <span className="icon">
-                  <i className="fas fa-sort" />
+    <>
+      {!data.length ? (
+        <p>There are no people matching the current search criteria</p>
+      ) : (
+        <table
+          data-cy="peopleTable"
+          className="table is-striped is-hoverable is-narrow is-fullwidth"
+        >
+          <thead>
+            <tr>
+              <th>
+                <span className="is-flex is-flex-wrap-nowrap">
+                  Name
+                  <a onClick={() => handleSortField('name')}>
+                    <span className="icon">
+                      <i
+                        className={cn(
+                          'fas',
+                          { 'fa-sort': sortBy !== 'name' },
+                          {
+                            'fa-sort-up': sortBy === 'name' && order !== 'desc',
```

Comment:
```suggestion
                            'fa-sort-up': sortBy === 'name' && !order,
```
U may try smth like this, same in other cases